doc/howtos/rdtraining/firstui.rst

```diff
@@ -0,0 +1,274 @@
+        <menuitem id="test_first_level_menu" name="First Level">
+            <menuitem id="test_model_menu_action" action="test_model_action"/>
+        </menuitem>
+    </menuitem>
+
+.. exercise:: Add the menus.
+
+    Create the ``estate_menus.xml`` file in the appropriate folder and define it in the
+    ``__manifest__.py`` file. Remember the sequential loading of the data files ;-)
+
+    Create the three levels of menus for the ``test_model_action`` action.
+
+Restart the server and **refresh the browser**. You should now see the menus, and you'll even
+be able to create your first real estate property advertisement!
+
+Fields, Attributes And View
+===========================
+
+.. note::
+
+    **Goal**: at the end of this section, the selling price should be read-only and the number
+    of bedrooms and the availaibity date is set by default. The latter is not copied when the
+    record is duplicated
+
+    .. image:: firstui/media/attribute_and_default.gif
+      :align: center
+      :alt: Interaction between model and view
+    
+    The reserved fields ``active`` and ``state`` are added to the ``estate.property`` model.
+
+Up to now, we were able to generate a default view for our real estate property advertisements.
+In many cases we want to fine tune the view. There are many fine-tunings possible in Odoo, but
+the very first step is usually to make sure that:
+
+- some fields have a default value
+- some fields are read-only
+- some fields are not copied when duplicating the record
+
+In our real estate business case, we would like the following:
+
+- The selling price should be read-only (it will be filled in automatically later)
+- The availability date and the selling price should not be copied when duplicating a record
+- The default number of bedrooms should be 2
+- The default availability date should be in 3 months
+
+Some New Attributes
+-------------------
+
+Before moving further on the view design, let's come back to our model definition. We saw that some
+attributes, such as ``required=True``, impact the table schema in the database. Other attributes
+will impact the view or provide default values.
+
+.. exercise:: Add new attributes to the fields
+
+  Find the appropriate attributes (see :class:`~odoo.fields.Field`) to:
+
+  - set the selling price as read-only
+  - prevent copying the value of the availability date and the selling price
+
+Restart the server and refresh the browser: you should not be able to set any selling price. When
+duplicating a record, the availability date should be empty.
+
+Default Values
+--------------
+
+Any field can be given a default value. In the field definition, add the option
+``default=X`` where ``X`` is either a Python literal value (boolean, integer,
+float, string), or a function taking a recordset and returning a value::
+
+    name = fields.Char(default="Unknown")
+    last_seen = fields.Datetime("Last Seen", default=lambda self: fields.Datetime.now())
+
```

Comment:
I will add another with Date

 date_start = fields.Date('Start Period', default=lambda self: fields.Date.today())

Reply:
Hum I would keep it this way, maybe we can add another field if the exercise is too difficult.

Reply:
It was to help a little with the exercise, in the exercise we use Date and not Datetime, and to help the newcomers to know that fields.Date.today() exists.

Reply:
Tip added:
```
Tip: this might help you: :meth:`~odoo.fields.Date.today`
```